test/react-markdown.test.js

```diff
@@ -893,3 +893,10 @@ test('should support formatting at the start of a GFM tasklist (GH-494)', () =>
   const expected = '<ul><li><input type="checkbox" readonly=""/><em>a</em></li></ul>'
   expect(actual).toEqual(expected)
 })
+
+test('should not crush on ending tag at start', () => {
+  const input = '<ruby><ruby></ruby></ruby>'
+  expect(() =>
+    renderHTML(<Markdown source={input} escapeHtml={false} astPlugins={[htmlParser()]} />)
+  ).not.toThrow()
+})
```

Comment:
(continuing from earlier conversation: https://github.com/remarkjs/react-markdown/pull/507/files#r525184845)

```suggestion
test('should not crash on weird `html-to-react` results', () => {
  const input = '<ruby><ruby></ruby></ruby>'
  const actual = renderHTML(<MarkdownWithHtml allowDangerousHtml children={input} />)
  // Note: this is not conforming to how browsers deal with it.
  const expected = '<p><ruby></ruby><ruby></ruby></p>'
  expect(actual).toEqual(expected)
})
```

Here is my suggestion: this uses the new properties (source -> children, escapeHtml -> allowDangerousHtml), and uses our component loaded with the html parser plugin. It adds a comment explaining why this test is needed.